src/harness/fourslash.ts

```diff
@@ -1797,6 +1797,10 @@ namespace FourSlash {
                 // this.languageService.getScriptLexicalStructure(fileName);
             }
 
+            if (this.currentCaretPosition < 0) {
+                this.currentCaretPosition = 0;

```

Comment:
It's an effect of this code to adjust caret position as we apply edits:
```ts
if (offsetStart <= this.currentCaretPosition) {
    this.currentCaretPosition += editDelta;
}
```
```editDelta``` can be negative. If you delete a span that contains the current caret position (perfectly valid), the caret position may end up going negative. 

There are alternative ways of fixing this (e.g. make sure the caret position remains positive _inside_ the for loop instead of fixing it afterwards) but I didn't see a compelling reason to prefer one over the other.